### PR TITLE
SharedTree: ChangeEncoder.decodeChange() should use 'Jsonable<T>'

### PR DESCRIPTION
**Pull request by DLehenbauer**
The 'Jsonable<T>' is another helper type that can catch some attempts to JSON serialize non-JSON serializable types.

It's typically used with type inference as follows:
```
function foo<T>(bar: Jsonable<T>) { ... }

foo(() => "not jsonable");    // Produces a compile time error.
```

See 'packages/runtime/datastore-definitions/src/jsonable.ts' for details.

### CODE REVIEW

File: packages/dds/tree/src/change-family/changeEncoder.ts [hunk 1]
```diff
@@ -4,9 +4,11 @@
  */
 
 import { bufferToString, IsoBuffer } from "@fluidframework/common-utils";
+import { Jsonable } from "@fluidframework/datastore-definitions";
 
 export abstract class ChangeEncoder<TChange> {
     public abstract encodeForJson(formatVersion: number, change: TChange): JsonCompatible;
```
Comment (DLehenbauer) [inline packages/dds/tree/src/change-family/changeEncoder.ts:10]: I'm curious about naming asymmetry between 'encode*For*Json' and 'decodeJson'.  Was that intentional?

**Comment by DLehenbauer** [comment]
@alex-pardes - Motivation for PR is to pass on knowledge of Jsonable<T>, which isn't easily discoverable.

File: packages/dds/tree/src/change-family/changeEncoder.ts (hunk 1 above)
Comment (alex-pardes) [inline packages/dds/tree/src/change-family/changeEncoder.ts:10]: I think this PR would be much more valuable if we can delete `JsonCompatible` by doing something like this:

```suggestion
export abstract class ChangeEncoder<TChange, Jsonable<TEncoded>> {
    public abstract encodeForJson(formatVersion: number, change: TChange): TEncoded;
```

Comment (alex-pardes) [reply packages/dds/tree/src/change-family/changeEncoder.ts:10]: @CraigMacomber actually wrote this file and picked these names, but the names seem fairly clear to me. I wouldn't object to something like `encodeAsJsonable` and `decodeFromJsonable`.

**Pull request closed without merging**